sig-scalability/processes/formal-scalability-processes.md

```diff
@@ -0,0 +1,136 @@
+
+At a high-level, these processes are (in the order in which we want to add them):
+
+- [Implementation (or) pre-submit phase](#implementation--pre-submit-phase):
+  - Add optional PR jobs for scalability (both medium and large scale)
+  - Add mandatory presubmit jobs for scalability (only medium scale - fast & healthy)
+- [Testing (or) post-submit phase](#testing--post-submit-phase):
+  - Add ability for crucial scalability jobs to block submit-queue
+- [Design (or) feature proposal phase](#design--feature-proposal-phase):
+  - Add a scalability review process for features/designs as needed
+
+Note that the above ordering is sorted from lighter to heavier weight processes. Further, after each phase, we would evaluate [our goal](#goal) for a month and exit (i.e not proceed with further phases) if it has already been met. So with the sorting and exit criteria combined, we’re ensuring that only a minimal but sufficient amount of red-tape is added to the development process.
+
+## Processes in Detail
+
+### Implementation / Pre-submit phase
+
+A good design does not always mean a good implementation. Besides, code changes are made that do not necessarily correspond to some feature. So we need checks in place for new code being checked-in to the repo, to ensure system scalability is not harmed. For this purpose, we would add two kinds of pre-submit scalability jobs running performance tests:
+
+#### Optional PR jobs
+
+These are scalability jobs that do not automatically run against PRs. They need 
+to be manually triggered using a bot command and are expected to be triggered by PR authors or reviewers (or delegated to sig-scalability if needed) if they believe the changes can affect scalability. These would further be of two kinds:
+
+- Medium-scale jobs\
+  ([relevant feature request](https://github.com/kubernetes/test-infra/issues/5677))\
+  Jobs that do not consume too many resources (light on quota-usage), not that high on cost and time and low-risk wrt outages, etc. These can be triggered by ~anybody. The following would be available here (can change over time):
+
+  - Kubemark-500 (consumes ~80 vCPUs)
+  - GCE-100 (consumes ~100 vCPUs)
+
+- Large-scale jobs\
+  ([relevant feature request](https://github.com/kubernetes/test-infra/issues/5051))\
+  Jobs that are much heavier wrt resources, time, cost and are high-risk. Their trigger would be access-restricted to a group of few googlers for those reasons. To begin with, the group would consist of key googlers from sig-scalability. Also these jobs need quota-checking before trigger and are to be used sparingly. The following would be available here (can change over time):
```

Comment:
We need to get past "key googlers" - are there no other sig-scale members tat could be authorized for this?

Reply:
Agree.. We should add more members from sig-scale. I'm just not super sure how granting access to our google-owned projects to outside folks works (as folks might need to check quota availability in the project before kicking them). Eventually we should try to get rid of such manual overhead somehow though.

Reply:
This should be solved at the test-infra level.

Before kicking the test, prow (or whatever else) should be checking how many other tests are running at a time, for how many we have quota in the project and depending on that accept or reject the request.

Reply:
Agree